### ISSUE TITLE
[Android 0.13.0-rc] JSApplicationIllegalArgumentException Error while updating property 'backgroundColor'

### ISSUE THREAD

**Issue by maraujop** [facebook/react-native#3630]
Hi,

I'm getting this traceback when I compile my react-native app for my Nexus 4 with Android 5.1.1 using react-native 0.13.0-rc. The apps crashes before starting showing this in logcat:

```
D/AndroidRuntime( 8467): Shutting down VM
E/AndroidRuntime( 8467): FATAL EXCEPTION: main
E/AndroidRuntime( 8467): Process: com.example, PID: 8467
E/AndroidRuntime( 8467): com.facebook.react.bridge.JSApplicationIllegalArgumentException: Error while updating property 'backgroundColor' of a view managed by: RCTView
E/AndroidRuntime( 8467):        at com.facebook.react.uimanager.ViewManagersPropertyCache$PropSetter.updateProp(ViewManagersPropertyCache.java:66)
E/AndroidRuntime( 8467):        at com.facebook.react.uimanager.ViewManager.updateProperties(ViewManager.java:47)
E/AndroidRuntime( 8467):        at com.facebook.react.uimanager.NativeViewHierarchyManager.createView(NativeViewHierarchyManager.java:182)
E/AndroidRuntime( 8467):        at com.facebook.react.uimanager.UIViewOperationQueue$CreateViewOperation.execute(UIViewOperationQueue.java:137)
E/AndroidRuntime( 8467):        at com.facebook.react.uimanager.UIViewOperationQueue$1.run(UIViewOperationQueue.java:574)
E/AndroidRuntime( 8467):        at com.facebook.react.uimanager.UIViewOperationQueue$DispatchUIFrameCallback.doFrameGuarded(UIViewOperationQueue.java:622)
E/AndroidRuntime( 8467):        at com.facebook.react.uimanager.GuardedChoreographerFrameCallback.doFrame(GuardedChoreographerFrameCallback.java:32)
E/AndroidRuntime( 8467):        at com.facebook.react.uimanager.ReactChoreographer$ReactChoreographerDispatcher.doFrame(ReactChoreographer.java:114)
E/AndroidRuntime( 8467):        at android.view.Choreographer$CallbackRecord.run(Choreographer.java:765)
E/AndroidRuntime( 8467):        at android.view.Choreographer.doCallbacks(Choreographer.java:580)
E/AndroidRuntime( 8467):        at android.view.Choreographer.doFrame(Choreographer.java:549)
E/AndroidRuntime( 8467):        at android.view.Choreographer$FrameDisplayEventReceiver.run(Choreographer.java:753)
E/AndroidRuntime( 8467):        at android.os.Handler.handleCallback(Handler.java:739)
E/AndroidRuntime( 8467):        at android.os.Handler.dispatchMessage(Handler.java:95)
E/AndroidRuntime( 8467):        at android.os.Looper.loop(Looper.java:135)
E/AndroidRuntime( 8467):        at android.app.ActivityThread.main(ActivityThread.java:5254)
E/AndroidRuntime( 8467):        at java.lang.reflect.Method.invoke(Native Method)
E/AndroidRuntime( 8467):        at java.lang.reflect.Method.invoke(Method.java:372)
E/AndroidRuntime( 8467):        at com.android.internal.os.ZygoteInit$MethodAndArgsCaller.run(ZygoteInit.java:903)
E/AndroidRuntime( 8467):        at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:698)
E/AndroidRuntime( 8467): Caused by: com.facebook.react.bridge.UnexpectedNativeTypeException: TypeError: expected dynamic type `int64', but had type `string'
E/AndroidRuntime( 8467):        at com.facebook.react.bridge.ReadableNativeMap.getInt(Native Method)
E/AndroidRuntime( 8467):        at com.facebook.react.uimanager.CatalystStylesDiffMap.getInt(CatalystStylesDiffMap.java:67)
E/AndroidRuntime( 8467):        at com.facebook.react.uimanager.ViewManagersPropertyCache$IntPropSetter.updateProperty(ViewManagersPropertyCache.java:91)
E/AndroidRuntime( 8467):        at com.facebook.react.uimanager.ViewManagersPropertyCache$PropSetter.updateProp(ViewManagersPropertyCache.java:63)
E/AndroidRuntime( 8467):        ... 19 more
W/ActivityManager(  567):   Force finishing activity 1 com.example/.MainActivity
```

I've tried installing the same app at the same time in my device and my emulator (nexus 4 Android 6). When I run it in my emulator sometime it crashes and maybe after 2 or 3 tries, sometimes on first try, I get this screen:

<img width="479" alt="captura de pantalla 2015-10-23 a las 13 51 09" src="https://cloud.githubusercontent.com/assets/252257/10691950/c12dfe60-798d-11e5-8171-5b81d6655c28.png">

Then when I reload my JS it starts working correctly, so I find this very inconsistent. 

I'm not sure if this a 0.13.0-rc regression or what is the culprit of this issue. I thought that maybe this was related to ReactNativeIcons which has a native bridge, but I removed all the corresponding code and the project from Android and it didn't help. Maybe I've got something wrong, but not sure what to look for.

Cheers,
Miguel

**Comment by oblador** [comment]
As of 0.12 (iirc) color processing moved to js, so you need to run `processColor` on any color strings first :-) If it's a third party lib causing this, then update that lib and file a bug report for that lib if it persists.

**Comment by maraujop** [comment]
Hi Joel,

Thank you for your fast answer. This is the first time I hear about `processColor`, I'm guessing this only affects to Android, because in iOS I'm not seeing any problem.

I've tried using `processColor` in every color I have in my project (I've reviewed 3rd party libs), but it is not working. I still get the traceback including this:

```
E/unknown:ViewManager( 2317): Error while updating prop backgroundColor
E/unknown:ViewManager( 2317): com.facebook.react.bridge.UnexpectedNativeTypeException: TypeError: expected dynamic type `int64', but had type `string'
E/unknown:ViewManager( 2317):   at com.facebook.react.bridge.ReadableNativeMap.getInt(Native Method)
E/unknown:ViewManager( 2317):   at com.facebook.react.uimanager.CatalystStylesDiffMap.getInt(CatalystStylesDiffMap.java:67)
E/unknown:ViewManager( 2317):   at com.facebook.react.uimanager.ViewManagersPropertyCache$IntPropSetter.updateProperty(ViewManagersPropertyCache.java:91)
E/unknown:ViewManager( 2317):   at com.facebook.react.uimanager.ViewManagersPropertyCache$PropSetter.updateProp(ViewManagersPropertyCache.java:63)
E/unknown:ViewManager( 2317):   at com.facebook.react.uimanager.ViewManager.updateProperties(ViewManager.java:47)
E/unknown:ViewManager( 2317):   at com.facebook.react.uimanager.NativeViewHierarchyManager.createView(NativeViewHierarchyManager.java:182)
E/unknown:ViewManager( 2317):   at com.facebook.react.uimanager.UIViewOperationQueue$CreateViewOperation.execute(UIViewOperationQueue.java:137)
E/unknown:ViewManager( 2317):   at com.facebook.react.uimanager.UIViewOperationQueue$1.run(UIViewOperationQueue.java:574)
E/unknown:ViewManager( 2317):   at com.facebook.react.uimanager.UIViewOperationQueue$DispatchUIFrameCallback.doFrameGuarded(UIViewOperationQueue.java:622)
E/unknown:ViewManager( 2317):   at com.facebook.react.uimanager.GuardedChoreographerFrameCallback.doFrame(GuardedChoreographerFrameCallback.java:32)
E/unknown:ViewManager( 2317):   at com.facebook.react.uimanager.ReactChoreographer$ReactChoreographerDispatcher.doFrame(ReactChoreographer.java:114)
E/unknown:ViewManager( 2317):   at android.view.Choreographer$CallbackRecord.run(Choreographer.java:765)
E/unknown:ViewManager( 2317):   at android.view.Choreographer.doCallbacks(Choreographer.java:580)
E/unknown:ViewManager( 2317):   at android.view.Choreographer.doFrame(Choreographer.java:549)
E/unknown:ViewManager( 2317):   at android.view.Choreographer$FrameDisplayEventReceiver.run(Choreographer.java:753)
E/unknown:ViewManager( 2317):   at android.os.Handler.handleCallback(Handler.java:739)
E/unknown:ViewManager( 2317):   at android.os.Handler.dispatchMessage(Handler.java:95)
E/unknown:ViewManager( 2317):   at android.os.Looper.loop(Looper.java:135)
E/unknown:ViewManager( 2317):   at android.app.ActivityThread.main(ActivityThread.java:5254)
E/unknown:ViewManager( 2317):   at java.lang.reflect.Method.invoke(Native Method)
E/unknown:ViewManager( 2317):   at java.lang.reflect.Method.invoke(Method.java:372)
E/unknown:ViewManager( 2317):   at com.android.internal.os.ZygoteInit$MethodAndArgsCaller.run(ZygoteInit.java:903)
E/unknown:ViewManager( 2317):   at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:698)
```

is there a way to know which RCTView is producing this? or maybe know anything else that can hint where to look?

Thanks again, cheers
Miguel

**Comment by oblador** [comment]
Hmm, maybe your npm version of react native is out of sync with the gradle version?

**Comment by maraujop** [comment]
Ok,

After several hours of fighting I've managed to fix. It wasn't `processColor` or an out of sync version of react-native-cli and react-native Android library. I decided yesterday to start a react-native project from scratch and move all my js code there and see if I had any compile issues, the result was unexpected, none.

So I tried doing a `diff -ru` between android directories (remember that my app was compiling correctly in both situations, however one was crashing before starting, the other not). The funny thing is that the only differences were the app names. So I renamed my app in my old directory project and compile it, and it started working. So I'm guessing that the name of my package, which contained the word `react` was causing trouble, not sure how or why.

In the meantime, I tried with my original name and setup in a virgin smartphone, to see If I had some cache problems in my device, but it also didn't work.

I'm not sure if I should leave the issue open or not.

Cheers,
Miguel
